### ISSUE TITLE
Dragging on the canvas

### ISSUE THREAD

**Issue by patryk84a** [mit-cml/appinventor-sources#2366]
**Describe the desired feature**

Could someone take a look at the canvas component? The idea is to modify the component so that some event returns all the coordinates from touching the canvas to moving to releasing. There is a "when.CanvasDraged" event but it shows the position only after 15 pixels. Link to the topic on the forum:

https://community.appinventor.mit.edu/t/asking-for-the-canvas-dragged-event/24264

**Comment by ewpatton** [comment]
Per the discussion on the forum, I think the best course of action is to make the drag threshold a user-settable property with the default being the existing threshold for backward compatibility. This could be a good starter project for someone looking to contribute to App Inventor.

**Comment by Himanshu2107** [comment]
Hi,
I am new to MIT App Inventor and would like to contribute. @ewpatton Can you assign this issue to me?

**Comment by ewpatton** [comment]
@Himanshu2107 Go for it!

**Comment by Himanshu2107** [comment]
I am getting an issue. 
The variable was declared in inner class `MotionEventParser` as 
`public static final int TAP_THRESHOLD = 15`
This value is used by `parse(MotionEvent event)` to determine whether it is a tap or a drag. To make this a user settable property  
I had put `TAP_THRESHOLD` as `private int tapThreshold = 15` in the main class `Canvas` and updated references to `TAP_THRESHOLD` to `tapThreshold`. After this I added the following getter and setter:

   `@SimpleProperty(category = PropertyCategory.BEHAVIOR,`
            `description = "Set the threshold number of pixels to differentiate a drag from a tap",`
            `userVisible = false`
   ` )`
   ` public int TapThreshold() {` 
         `return tapThreshold;` 
   `}`

    @SimpleProperty
    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_INTEGER)
    public void TapThreshold(int threshold) {
        this.tapThreshold = threshold;
    }

I also added TapThreshold(15) in the constructor.

As modifying this property does not changes the apperance of either designer or android, no other code is changed or added.

Here is the issue I am getting. After connecting my phone through the companion, whenever the value of TapThreshold is different from 15 in the designer, I get a runtime error as popup on my browser screen:

`Error from Companion: invoke: no method named 'TapThreshold' in class com.google.appinventor.components.runtime.Canvas` 
 
If value is changed from 15 the error occurs. Changing the value back to 15 makes the error disappear. 

@ewpatton  Can you help me with this?

**Comment by ewpatton** [comment]
App Inventor only sends a property if it is different from the default. You're probably running this with the production companion. Once you've built all of App Inventor (or at least run `ant PlayApp`), you will need to install the APK produced at `build/buildserver/MIT AI2 Companion.apk` on your phone in place of the version you get from the Play Store. In the `@DesignerProperty` annotation, make sure you also set `defaultValue = "15"`.

**Comment by Himanshu2107** [comment]
Yes, that was it. By installing the new app, the changes are running perfectly.

**Comment by Himanshu2107** [comment]
Yes, that was it. By installing the new app, the changes are running perfectly.

**Comment by Himanshu2107** [comment]
Before creating a pull request, I should first pull from upstream ucr and then push to my fork. After that I should create a pull request. Right?

**Comment by ewpatton** [comment]
I think your PR was fine with the exception of the fact that you merged `master` into your branch based on `ucr`. We typically only do that step just before a companion release. As long as you keep all of your component change logic based off `ucr` GitHub should resolve things correctly. You could try reverting 373c59d on your branch and then pushing that up to see if it will correct. If not, create a clean branch off `ucr` and cherry-pick 94fdfa2 onto it.

**Comment by Himanshu2107** [comment]
The branch I worked on was created off master but it should have been created off ucr. So I had to create a new branch off ucr and make changes there. GitHub is now resolving things correctly. Thank You @ewpatton for your support.

**Comment by Himanshu2107** [comment]
@ewpatton Issue is resolved, please close.

**Comment by ewpatton** [comment]
@Himanshu2107 Since `ucr` hasn't been merged into `master` yet I've marked it as pending component release. When we do the release we'll go through and close the pending issues.

**Comment by Himanshu2107** [comment]
Got it.
